File: backend/Contracts/DF.Contracts/Gateway/Responses/AccountResponse.cs
using System;
using System.Text.Json.Serialization;

namespace DF.Contracts.Gateway.Responses;

[JsonPolymorphic(TypeDiscriminatorPropertyName = "$type")]
[JsonDerivedType(typeof(CustomerAccountResponse), "customer")]
[JsonDerivedType(typeof(BusinessAccountResponse), "business")]
[JsonDerivedType(typeof(CourierAccountResponse), "courier")]
public abstract record AccountResponse
{
    public string? Id { get; init; }
    public string? UserId { get; init; }
    public string AccountType { get; init; } = default!;
    public string? ImageUrl { get; init; }
}


public record CustomerAccountResponse : AccountResponse
{
    public string? PhoneNumber { get; init; }
    public string? Name { get; init; }
    public string? Surname { get; init; }
    public string? Address { get; init; }
}


public record BusinessAccountResponse : AccountResponse
{
    public string Name { get; init; } = default!;
    public string? Description { get; init; }
    public string? StripeAccountId { get; init; }
    public bool? StripeChargesEnabled { get; init; }
    public bool? StripePayoutsEnabled { get; init; }
    public string? StripeRequirementsDue { get; init; }
    public DateTime? StripeOnboardedAt { get; init; }
}


public record CourierAccountResponse : AccountResponse
{
    public string? PhoneNumber { get; init; }
    public string? Name { get; init; }
    public string? Surname { get; init; }
    public string? Address { get; init; }
    public string? Description { get; init; }
}

File: frontend/food-delivery-platform/src/models/responses/accounts/AccountResponse.ts
// =========================
// Base
// =========================
export interface AccountResponseBase {
    id?: string
    userId?: string
    accountType: string
    imageUrl?: string

    /** Polymorphic discriminator from backend */
    $type: "customer" | "business" | "courier"
}
export interface CustomerAccountResponse extends AccountResponseBase {
    $type: "customer"

    phoneNumber?: string
    name?: string
    surname?: string
    address?: string
}
export interface BusinessAccountResponse extends AccountResponseBase {
    $type: "business"

    name: string
    description?: string
    stripeAccountId?: string
    stripeChargesEnabled?: boolean
    stripePayoutsEnabled?: boolean
    stripeRequirementsDue?: string
    stripeOnboardedAt?: string // ⬅ ISO string (DateTime → string)
}
export interface CourierAccountResponse extends AccountResponseBase {
    $type: "courier"

    phoneNumber?: string
    name?: string
    surname?: string
    address?: string
    description?: string
}

export type AccountResponse =
    | CustomerAccountResponse
    | BusinessAccountResponse
    | CourierAccountResponse
``